tests/unit/amo/pages/TestSearchPage.js

```diff
@@ -1,375 +1,824 @@
-  shallowUntilTarget,
+  dispatchSearchResults,
+  fakeAddon,
+  getElement,
+  getElements,
+  getSearchErrorHandlerId,
+  renderPage as defaultRender,
+  screen,
 } from 'tests/unit/helpers';
 
 describe(__filename, () => {
   let store;
-
-  function render({
-    location = createFakeLocation({ query: { page: 2, q: 'burger' } }),
-    ...props
-  } = {}) {
-    return shallowUntilTarget(
-      <SearchPage location={location} store={store} {...props} />,
-      SearchPageBase,
-    );
-  }
+  const lang = 'en-US';
+  const clientApp = CLIENT_APP_FIREFOX;
+  const defaultLocation = `/${lang}/${clientApp}/search/`;
 
   beforeEach(() => {
-    store = dispatchClientMetadata().store;
-  });
-
-  it('renders a SearchPage', () => {
-    const root = render();
-
-    expect(root.find(Search)).toHaveLength(1);
-  });
-
-  it("doesn't duplicate the clientApp in the URL in the queryParams", () => {
-    const root = render({
-      location: createFakeLocation({ query: { page: 3, q: 'fries' } }),
-    });
-
-    expect(root.find(Search).prop('paginationQueryParams')).toEqual({
-      page: 3,
-      q: 'fries',
-    });
-  });
-
-  it('sets the paginationQueryParams from filters', () => {
-    const root = render({
-      location: createFakeLocation({
-        query: {
-          page: 2,
-          q: 'burger',
-          tag: 'firefox57',
-        },
-      }),
-    });
-
-    expect(root.find(Search)).toHaveProp('paginationQueryParams', {
-      page: 2,
-      q: 'burger',
-      tag: 'firefox57',
-    });
+    store = dispatchClientMetadata({ clientApp, lang }).store;
   });
 
-  it('sets the filters from the location', () => {
-    const page = 2;
-    const q = 'burger';
-    const tag = 'firefox57';
-    const root = render({
-      location: createFakeLocation({
-        query: { page, q, tag },
-      }),
-    });
+  function render({ history, location = defaultLocation, term } = {}) {
+    const renderOptions = {
+      history:
+        history ||
+        createHistory({
+          initialEntries: [term ? `${defaultLocation}?q=${term}` : location],
+        }),
+      store,
+    };
+    return defaultRender(renderOptions);
+  }
 
-    expect(root.find(Search)).toHaveProp('filters', { page, query: q, tag });
+  const renderWithResults = ({
+    count = 5,
+    filterProps = {},
+    history,
+    page = '1',
+    pageSize = 2,
+    term,
+  } = {}) => {
+    const addons = Array(pageSize).fill(fakeAddon);
+    const filters = { page, ...filterProps };
+    if (term) {
+      filters.query = term;
+    }
+    dispatchSearchResults({ addons, count, filters, pageSize, store });
+    let location = `${defaultLocation}?page=${page}`;
+    if (term) {
+      location = `${location}&q=${term}`;
+    }
+    for (const key of Object.keys(filterProps)) {
+      location = `${location}&${key}=${filterProps[key]}`;
+    }
+    render({ history, location });
+  };
+
+  it("doesn't duplicate the clientApp in the URL in the queryParams", async () => {
```

Comment:
It doesn't seem like this is testing anything specific that wouldn't be caught by the other tests?

Reply:
I think you're right. Moving these tests around/changing the way we set them up ends up making some of them irrelevant, and I think this is one. I'll remove it.